selfservice/flow/verification/handler.go

```diff
@@ -248,15 +249,51 @@ func (h *Handler) fetch(w http.ResponseWriter, r *http.Request, _ httprouter.Par
 
 	if req.ExpiresAt.Before(time.Now().UTC()) {
 		if req.Type == flow.TypeBrowser {
+			reason := "The verification flow has expired. Redirect the user to the verification flow init endpoint to initialize a new verification flow."
+
+			requestURL := urlx.AppendPaths(h.d.Config(r.Context()).SelfPublicURL(), RouteInitBrowserFlow)
+			query := requestURL.Query()
+			if req.GetReturnTo() != "" {
+				query.Set("return_to", req.GetReturnTo())
+			}
+			requestURL.RawQuery = query.Encode()
+			RequestUrl, err := url.PathUnescape(requestURL.String())
+
+			if err != nil {
+				h.d.Writer().WriteError(w, r, errors.WithStack(x.ErrGone.
+					WithReason(reason).
+					WithDetail("redirect_to", requestURL.String())))
+				return
+			}
+
+			h.d.Writer().WriteError(w, r, errors.WithStack(x.ErrGone.
+				WithReason(reason).
+				WithDetail("redirect_to", RequestUrl)))
+			return
+		}
+
+		reason := "The verification flow has expired. Call the verification flow init API endpoint to initialize a new verification flow."
+
+		requestURL := urlx.AppendPaths(h.d.Config(r.Context()).SelfPublicURL(), RouteInitAPIFlow)
+		query := requestURL.Query()
+		if req.GetReturnTo() != "" {
+			query.Set("return_to", req.GetReturnTo())
+		}
+		requestURL.RawQuery = query.Encode()
+		RequestUrl, err := url.PathUnescape(requestURL.String())
+
+		if err != nil {
 			h.d.Writer().WriteError(w, r, errors.WithStack(x.ErrGone.
-				WithReason("The verification flow has expired. Redirect the user to the verification flow init endpoint to initialize a new verification flow.").
-				WithDetail("redirect_to", urlx.AppendPaths(h.d.Config(r.Context()).SelfPublicURL(), RouteInitBrowserFlow).String())))
+				WithReason(reason).
+				WithDetail("api", requestURL.String())))
```

Comment:
Why not
```suggestion
				WithDetail("original_return_to", req.GetReturnTo())))
```
and use the same mechanism (SDK/helper function/...) in the UI to init the flow as you did originally?

Reply:
Hey @zepatrik, I'm a bit unfamiliar with the api side. Would this affect any other part of the system or only the people whom are consuming the api error? I'll commit the changes for now and revert if needed.

Reply:
So it now looks like this for the api flow where the verification link has expired:
```go
reportErr := x.ErrGone.WithReason("The verification flow has expired. Call the verification flow init API endpoint to initialize a new verification flow.")

h.d.Writer().WriteError(w, r, errors.WithStack(reportErr.WithDetail("original_return_to", req.GetReturnTo())))
return
```

Reply:
The thing you want to have is preserve the return URL parameter. I'm sure there are some people who don't want that. And if we ever add more params to the flow init URLs, we will have to decide whether to preserve them or not. Leaving this option to the application will make it a lot more flexible and straight-forward later on.
That's my reasoning behind not sending the whole encoded URL :wink:

Reply:
ahh yeah that makes perfect sense :)